File: snippets/php/add_bulk_tags.php
require_once('/path/to/MailchimpMarketing/vendor/autoload.php');

$mailchimp = new MailchimpMarketing\ApiClient();

$mailchimp->setConfig([
    'apiKey' => 'YOUR_API_KEY',
    'server' => 'YOUR_SERVER_PREFIX'
]);

$list_id = "YOUR_LIST_ID";
$tag_id = "YOUR_TAG_ID";

try {
    $response = $mailchimp->lists->batchSegmentMembers(["body" => ["dolly.parton@example.com", "rihanna@example.com"]], $list_id, $tag_id);
    echo "Successfully tagged {$response["total_added"]} contacts";
} catch (MailchimpMarketing\ApiException $e) {
    echo $e->getMessage();
}

File: snippets/python/add_bulk_tags.py
import mailchimp_marketing as MailchimpMarketing
from mailchimp_marketing.api_client import ApiClientError

LIST_ID = "YOUR_LIST_ID"
TAG_ID = "YOUR_TAG_ID"

client = MailchimpMarketing.Client()
mailchimp.set_config({
  "api_key": "YOUR_API_KEY",
  "server": "YOUR_SERVER_PREFIX"
})
try:
    response = client.lists.batch_segment_members({
        "members_to_add": [
            "dolly.parton@example.com",
            "rihanna@example.com",
        ]
    }, LIST_ID, TAG_ID)
    print("client.ping.get() response: {}".format(response))
except ApiClientError as error:
    print("An exception occurred: {}".format(error.text))